src/HealthChecks.Rabbitmq/RabbitMQHealthCheck.cs

```diff
@@ -9,19 +9,22 @@ namespace HealthChecks.RabbitMQ;
 /// </summary>
 public class RabbitMQHealthCheck : IHealthCheck
 {
-    private static readonly ConcurrentDictionary<string, IConnection> _connections = new();
+    private static readonly ConcurrentDictionary<RabbitMQOptions, IConnection> _connections = new();
 
     private IConnection? _connection;
     private readonly IConnectionFactory? _factory;
+    private readonly RabbitMQOptions _options;
 
     public RabbitMQHealthCheck(IConnection connection)
     {
         _connection = Guard.ThrowIfNull(connection);
+        _options = new RabbitMQOptions(new Uri(connection.Endpoint.ToString()));
```

Comment:
When I was talking about options I mean passing options instance into single ctor. Options class contains all needed properties - factory, url, connection, etc. ? See https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.AzureServiceBus/Configuration/AzureEventHubHealthCheckOptions.cs

Reply:
The original problem was obtaining SSL config from an IFactory, so it would not be possible to generate a complete RabbitMQHealthCheckConfig from an IFactory. On the other hand we can build an IFactory from an URI and SSL config, so we could use the IFactory instead of RabbitMQHealthCheckConfig.

For what concerns passing a Connection we cannot do much: from it we cannot obtain neither an IFactory, neither a RabbitMQHealthCheckConfig.